src/main/java/io/rpg/controller/Controller.java

```diff
@@ -126,6 +128,11 @@ public void onKeyboardEvent(KeyboardEvent event) {
     }
   }
 
+  private void battle(GameObject opponent){
+    Battle battle = new Battle(currentModel.getPlayer(), opponent, 10);
+    popupController.openTextPopup(battle.action(), getWindowCenterX(), getWindowCenterY());
```

Comment:
Not essential, but TextImagePopup could be used here instead, with an image of the opponent, or alternatively a smiley/sad face corresponding to the result.